katsdpdata/prod_handler.py

```diff
@@ -483,6 +501,125 @@ def rdb_file_prefix(self):
+        This key is also used as the product ID
+        """
+        name = os.path.split(self.product_path.rstrip('/'))[1]
+        return f'{name}-sdp-l0'
+
+    def _get_product_prefix(self):
+        return self._get_key_from_product_path().replace('-sdp-l0', '')
+
+    def discover_trawl_files(self):
+        """Discover this products trawl files.
+        Only transfer once complete token is available, otherwise []."""
+        super()._discover_trawl_files('*.rdb', '*.writing.rdb', 'complete')
+        if not self.complete:
+            self.file_matches = []
+
+    def metadata_transfer_complete(self):
+        """Update metadata when transfer completes"""
+        mh = self.mh()
+        met = mh.get_prod_met()
+        met = mh.add_inferred_ref_datastore(met)
+        # TODO: We are now setting the state to received again, which isn't
+        # TODO: necessary, perhaps we should rather set the time when we set
+        # TODO: the state in the state change method?
+        met = mh.set_product_received(met)
+        met_bucket = self.get_bucket_stats()
+        mh.add_bucket_stats(met, met_bucket)
+
+    def set_rdb_metadata(self, available_refs, original_refs):
+        #set_trace()
+        """Ingest a product into the archive. This includes extracting and uploading
+        metadata and then moving the product into the archive.
+
+        Parameters
+        ----------
+        trawl_dir: string : full path to directory to trawl for ingest product.
+        prod_id: string : unique id for the product.
+        original_refs : list : list of product file(s).
+        prod_met_extractor: class : a metadata extractor class.
+        solr_url: string : sorl endpoint for metadata queries and upload.
+        boto_dict: dict
+
+        Returns
+        -------
+        met : dict : a metadata dictionary with uploaded key:value pairs.
+        """
+        try:
+            pm_extractor = file_mime_detection(available_refs[0])
+            pm_extractor.extract_metadata()
+        except Exception as err:
+            err.bucket_name = self.bucket_name()
+            raise err
+        # Either get the product met or at least create the core meta data
+        mh = self.mh(pm_extractor.product_type)
+        met = mh.get_prod_met(self.key)
+        if not met:
+            met = mh.create_core_met()
+        if "CAS.ProductTransferStatus" in met and met[
+                "CAS.ProductTransferStatus"] == "RECEIVED":
+            err = MetExtractorException(
+                "%s marked as RECEIVED, while trying to create new product.",
+                mh.product_id)
+            err.bucket_name = self.bucket_name()
+            raise err
+        met_original_refs = list(original_refs)
+        met_original_refs.insert(0, os.path.dirname(
+            os.path.commonprefix(original_refs)))
+        met = mh.add_ref_original(met, met_original_refs)
+        mh.add_prod_met(met, pm_extractor.metadata)
+
+    def metadata_when_created(self):
```

Comment:
unchanged from base class remove.

Reply:
If I don't have a return/pass statement in this method for pruned products, this will throw an error. I'm actually overring the method.

Reply:
ah, yes, but then the rest of the code should go...

Reply:
okay,I see what you mean.